### PR TITLE
test: fix component test test_configdrive.py 

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes some issues in test_configdrive.py and moves it from component test to smoke test.

(sleep 30s as vm crashes after migration in ubuntu 20.04)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@nvazquez moved to 4.16.1.0

**Comment by sureshanaparti** [comment]
@weizhouapache Is this PR ready for review?

**Comment by weizhouapache** [comment]
> @weizhouapache Is this PR ready for review?

@sureshanaparti not yet. facing some issue with template registration in smoke test environments. I will look into it.

**Comment by sureshanaparti** [comment]
> > @weizhouapache Is this PR ready for review?
> 
> @sureshanaparti not yet. facing some issue with template registration in smoke test environments. I will look into it.

ok, thanks for the update @weizhouapache

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

File: test/integration/smoke/test_configdrive.py [hunk 1]
```diff
@@ -1089,10 +1090,8 @@ def update_and_validate_sshkeypair(self, vm, public_ip=None):
 
         vm.details = vm_new_ssh.details
 
-        # reset SSH key also resets the password.
-        self._decrypt_password(vm)
-
-        vm.password_test = ConfigDriveUtils.PasswordTest(vm=vm)
+        # reset SSH key also remove the password (see https://github.com/apache/cloudstack/pull/4819)
```
Comment (DaanHoogland) [inline test/integration/smoke/test_configdrive.py:1093]:
```suggestion
        # reset SSH key also removes the password (see https://github.com/apache/cloudstack/pull/4819)
```

**Review by DaanHoogland — APPROVED** [review]
clgtm (small comment spello suggestion)

**Comment by sureshanaparti** [comment]
> Trillian test result (tid-2688) Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7 Total time taken: 30919 seconds Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr5528-t2688-kvm-centos7.zip Smoke tests completed. 91 look OK, 1 have errors Only failed tests results shown below:
> 
> Test	Result	Time (s)	Test File
> runTest	`Error`	0.00	test_configdrive.py
> runTest	`Error`	0.00	test_configdrive.py

@weizhouapache can you check these test failures.

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test suse15 kvm-suse15

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test suse15 kvm-suse15

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
code looks good to me but it still doesn't seem to work @weizhouapache . Do you have any idea yet?

> Trillian test result (tid-2705) Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7 Total time taken: 63548 seconds Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr5528-t2705-kvm-centos7.zip Smoke tests completed. 91 look OK, 1 have errors Only failed tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_configdrive_isolated_network 	`Error` 	5213.49 	test_configdrive.py
> test_configdrive_shared_network 	`Error` 	1352.46 	test_configdrive.py

**Comment by weizhouapache** [comment]
> code looks good to me but it still doesn't seem to work @weizhouapache . Do you have any idea yet?
> 
> > Trillian test result (tid-2705) Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7 Total time taken: 63548 seconds Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr5528-t2705-kvm-centos7.zip Smoke tests completed. 91 look OK, 1 have errors Only failed tests results shown below:
> > Test 	Result 	Time (s) 	Test File
> > test_configdrive_isolated_network 	`Error` 	5213.49 	test_configdrive.py
> > test_configdrive_shared_network 	`Error` 	1352.46 	test_configdrive.py

@DaanHoogland I am checking

**Comment by weizhouapache** [comment]
@blueorangutan test keepEnv

**Comment by weizhouapache** [comment]
just run the test on a env with pr 5769, it looks good.
```
[root@pr5769-t2700-kvm-centos7-marvin ~]# cat /marvin//MarvinLogs//06MT81/results.txt 
Test Configdrive as provider for isolated Networks ... === TestName: test_configdrive_isolated_network | Status : SUCCESS ===
ok
Test Configdrive as provider for isolated Networks ... === TestName: test_configdrive_isolated_network_hypervisor_hostname_exposed | Status : SUCCESS ===
ok
Test Configdrive as provider for shared Networks ... === TestName: test_configdrive_shared_network | Status : SUCCESS ===
ok
Test Configdrive for VPC Networks ... === TestName: test_configdrive_vpc_network | Status : SUCCESS ===
ok
Test Configdrive for VPC Networks ... === TestName: test_configdrive_vpc_network_verify_metadata | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 5 tests in 6192.782s

OK
```


rekick the test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@sureshanaparti @DaanHoogland 
test passed. this is ready for review.
However, I wonder if neccesary to move it to smoke test (run the tests in 103 minutes in my testing)

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Review by sureshanaparti — APPROVED** [review]

**Comment by DaanHoogland** [comment]
> However, I wonder if neccesary to move it to smoke test (run the tests in 103 minutes in my testing)

Wei, can we extract a minimal set into a smoke test and leave the rest in components?

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**